Adding api documentation issue/43 (#82)
OpenBAS-Platform/docs: docs/usage/rest-api.md

# REST API

> OpenBAS provides a REST API, allowing users to perform various actions programmatically. The API enables users to interact with OpenBAS's functionality and data, offering a powerful tool for automation, integration, and customization. All actions that can be performed through the platform's graphical interface are also achievable via the API.

Reviewer: ```suggestion
OpenBAS provides a REST API, allowing users to perform various actions programmatically. The API enables users to interact with OpenBAS's functionality and data, offering a powerful tool for automation, integration, and customization. Any action available through the platform's graphical interface can also be executed via the API.
```

## Authentication

> Access to the OpenBAS API requires authentication using standard authentication mechanisms. For authentication, users need to include the following headers in their API requests:

Reviewer: ```suggestion
Accessing the OpenBAS API requires authentication through standard mechanisms. To authenticate, users must include the following headers in their API requests:
```
